ha-jobs-core/src/main/scala/de/kaufhof/hajobs/JobStatusRepository.scala

```diff
@@ -79,22 +80,30 @@ class JobStatusRepository(session: Session,
 
   /**
    * Finds the latest job status entries for all jobs.
+   * Every JobState is loaded with a single select statement
+   * It is recommened to use partition keys if possible and avoid
+   * IN statement in WHERE clauses, therefore we prefer to execute
+   * more than one select statement
    */
-  def getAllMetadata(readWithQuorum: Boolean = false)(implicit ec: ExecutionContext): Future[List[JobStatus]] = {
-    import scala.collection.JavaConversions._
-
-    val selectStmt = select().all().from(MetaTable)
+  def getLatestMetadata(readwithQuorum: Boolean = false)(implicit ec: ExecutionContext): Future[List[JobStatus]] = {
 
-    if (readWithQuorum) {
-      // setConsistencyLevel returns "this", we do not need to reassign
-      selectStmt.setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM)
+    def selectStmt(jobType: JobType, withQuorum: Boolean) = {
```

Comment:
The local function does not need the "withQuorum" parameter. readWithQuorum from the outer function can just be used.